src/pal/src/loader/module.cpp

```diff
@@ -1339,29 +1342,59 @@ static HMODULE LOADLoadLibrary(LPCSTR shortAsciiName, BOOL fDynamic)
     // See if file can be dlopen()ed; this should work even if it's already loaded
     {
         // See GetProcAddress for an explanation why we leave the PAL.
-        PAL_LeaveHolder holder;
-        dl_handle = dlopen(shortAsciiName, RTLD_LAZY | RTLD_NOLOAD); 
-        if (!dl_handle)
+        PAL_LeaveHolder holder; 
+        
+        // Try to dlopen the module with appropriate variations on the name.
+        // P/Invokes are often declared with shortened versions of the real library
+        // name, such as without the extension or without the "lib" prefix.
+        // We try to open with such variations on the original.
+        int endsWithSuffix = FALSE, startsWithPrefixOrIsAbsolute = FALSE;
+        size_t shortAsciiNameLen = 0;
+        int i;
+        for (i = 1; i <= 4; i++) // four cases: original, with prefix, with suffix, with both prefix and suffix
         {
-            dl_handle = dlopen(shortAsciiName, RTLD_LAZY);
-        }
+            const char* filename = NULL;
+            switch (i)
+            {
+                case 1: // original name
+                    filename = shortAsciiName;
+                    break;
+
+                case 2: // with prefix prepended
```

Comment:
> Are we agreed on that?

I agree with that list, those are common cases that should catch a good chunk of p/invokes.

I'm not so sure about `prefixname`, I agree with @OtherCrashOverride that we should push people to stick with `name` and not put .dll/.so into the dllimport. I'd prefer removing `prefixname` for now to avoid people relying on this behavior.